lib/resty/limit/req.lua

```diff
@@ -160,17 +163,26 @@ function _M.limit(cfg)
 
     rate = floor((tonumber(rate) or 1) * 1000 / scale)
 
-    local res, err = redis_lookup(conn, zone, key, rate, interval, burst)
+    local res, err = redis_lookup(conn_slave, zone, key, rate, interval, burst)
     if res and (res[1] == _M.BUSY or res[1] == _M.FORBIDDEN) then
         if res[1] == _M.BUSY then
-            ngx.log(log_level, 'limiting requests, excess ' ..
-                        res[2]/1000 .. ' by zone "' .. zone .. '"')
+            ngx.log(log_level, "excess requests " ..
+                    zone .. ":" .. key .. " - " .. res[2]/1000 )
         end
         return
     end
-
     if not res and err then
-        ngx.log(ngx.WARN, "redis lookup err: ", err)
+        ngx.log(ngx.WARN, "redis lookup error: ", err)
+        return _M.OK
+    end
+
+    for host, conn in pairs(conn_masters) do
```

Comment:
Verifying correct writes to all redis master hosts with more than 2 hosts is getting tricky.
If with 2 hosts I can skip second one if first one failed, then with 3 or more hosts I have to:
a) just write to all of them and hope it worked on all of them
b) write a revert function (hell no)

I chose a), that's why I'm failing (and returning OK) above, if connecting to at least one master host failed.